src/error.rs

```diff
@@ -9,68 +9,85 @@ use thiserror::Error;
 #[derive(Error, Debug)]
 #[non_exhaustive]
 pub enum GeoArrowError {
+    /// Incorrect type was passed to an operation.
     #[error("Incorrect type passed to operation: {0}")]
     IncorrectType(Cow<'static, str>),
 
     /// Returned when functionality is not yet available.
     #[error("Not yet implemented: {0}")]
     NotYetImplemented(String),
 
+    /// General error.
     #[error("General error: {0}")]
     General(String),
 
     /// Whenever pushing to a container fails because it does not support more entries.
+    ///
     /// The solution is usually to use a higher-capacity container-backing type.
     #[error("Overflow")]
     Overflow,
 
+    /// [ArrowError]
```

Comment:
As a general statement, I don't know whether having a separate GeoArrowError variant for all of these is the right approach, or whether having a single `ExternalError(Box<dyn Error>)` is more general/better. Maybe since geoarrow-rs is a library, having them split up is better for consumers because they know exactly what the error type was?

Reply:
Yup, I much prefer having the separate enum variants w/ `#[from]`. It can be a bit of boilerplate choreing on initial library standup, but after that I don't find it's too bad.

From the downstream perspective (e.g. **stac-geoparquet**) the varients are better IMO, because that lets me (e.g.) catch a `GeoArrowError::SerdeJsonError` and do some sort of special JSON processing (e.g. looking for nd-json or something) without having to do any dynamic casting.